src/core-rpc/client.ts

```diff
@@ -208,6 +268,60 @@ export class StacksCoreRpcClient {
     };
   }
 
+  async multicastTransaction(
+    serializedTx: Buffer
+  ): Promise<{
+    txId: string | undefined;
+    errors: {
+      endpoint: string;
+      error: Error;
+    }[];
+  }> {
+    const extraEndpoints = await getExtraTxPostEndpoints();
+    if (!extraEndpoints) {
+      const result = await this.sendTransaction(serializedTx);
+      return {
+        txId: result.txId,
+        errors: [],
+      };
+    }
+    const endpoints = [
+      // The primary proxy endpoint (the http response from this one will be returned to the client)
+      this.createUrl('v2/transactions'),
+      ...extraEndpoints.map(e => this.createUrl('v2/transactions', { endpoint: e })),
+    ];
+    // Remove dupes
+    for (let i = endpoints.length - 1; i >= 0; i--) {
+      for (let f = endpoints.length - 1; f >= 0; f--) {
+        if (f !== i && endpoints[i] === endpoints[f]) {
+          endpoints.splice(f, 1);
+        }
+      }
+    }
```

Comment:
To avoid this O(n^2) loop, you could feed the endpoints to a `Map<string, bool>` (taking the keys afterwards) or a `Set` from the start.

I'm also not sure on what JS would do if you modify the `endpoints` array while you're iterating it. Some other languages I've come across freak out with an OOB exception when this happens.

Reply:
Map would be better except for not preserving order (the first endpont is treated special). But yeah I'll look for a better way to do this, maybe a Set somehow.